File: auto-docs/executables/r/simple_contour.r
# Learn about API authentication here: {{BASE_URL}}/r/getting-started
# Find your api_key here: {{BASE_URL}}/settings/api

library(plotly)

size <- 100
x <- seq(-2*pi, 2*pi, length=size)
y <- seq(-2*pi, 2*pi, length=size)
z <- matrix(0, size, size)
for(i in 1:size) {
    for(j in 1:size) {
        r2 <- x[i]^2 + y[j]^2
        z[i, j] <- sin(x[i])*cos(y[j])*sin(r2)/log(r2+1)
    }
}

py <- plotly(username='TestBot', key='r1neazxo9w')
data <- list(
  list(
    z = z, 
    x = x, 
    y = y, 
    type = "contour"
  )
)
response <- py$plotly(data, kwargs=list(filename="simple-contour", fileopt="overwrite"))
url <- response$url

File: auto-docs/executables/python/simple_contour.py
# Learn about API authentication here: plot.ly/python/getting-started
# Find your api_key here: plot.ly/settings/api

import plotly.plotly as py
from plotly.graph_objs import *
py.sign_in('TestBot', 'r1neazxo9w')

import numpy as np

size = 100
x = np.linspace(-2*np.pi, 2*np.pi, size)
y = np.linspace(-2*np.pi, 2*np.pi, size)
z = np.empty((size, size))
for i, xi in enumerate(x):
    for j, yj in enumerate(y):
        r2 = (xi**2+yj**2)
        z[i][j] = np.sin(xi)*np.cos(yj)*np.sin(r2)/(np.log(r2+1))

data = Data([
    Contour(
        z=z,
        x=x,
        y=y
    )
])
plot_url = py.plot(data, filename='simple-contour', auto_open=False)
